cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -326,11 +381,12 @@ public void deleteAll(Range keyRange) throws InvalidFieldException, IOException
   public long count(Collection<Range> keyRanges) throws IOException {
     LOG.trace("Table {}: count with ranges {}", schema.getTableId(), keyRanges);
     long count = 0;
-    for (Range keyRange: keyRanges) {
-      try (Scanner scanner = getScanner(keyRange)) {
```

Comment:
The original `count(Collection<Range> keyRanges)` is problematic that it could return repeated counts if the ranges overlap with each other